README.md

```diff
@@ -1,17 +1,121 @@
+resources.
 
-## How to help
+To provide this functionality, Ingress Operator deploys and manages the
+[OpenShift router](https://github.com/openshift/router) — a
+[HAProxy-based](https://www.haproxy.com) Kubernetes [ingress
+controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers).
 
-See [HACKING.md](HACKING.md) for development topics.
+## Install
+
+Ingress Operator is a core feature of
+[OpenShift](https://github.com/openshift/installer) and is enabled out of the
+box.
+
+Every new OpenShift installation has an `ingresscontroller` named `default`
+which can be customized, replaced, or supplemented with additional ingress
+controllers. To view the default ingress controller, use the `oc` command:
+
+```shell
+$ oc describe --namespace=openshift-ingress-operator ingresscontroller/default
+```
+
+## Manage
+
+Ingress Operator provides the OpenShift [ingresscontroller API](). Create and
+edit `ingresscontroller.operator.openshift.io` resources to manage ingress
+controllers.
+
+Interact with ingress controllers using the `oc` command. Every ingress
+controller lives in the `openshift-ingress-operator` namespace.
+
+To scale an ingress controller:
+
+```shell
+$ oc scale --replicas=1 --namespace=openshift-ingress-operator ingresscontroller/<name>
+```
+
+## Customize
+
+Create new `ingresscontroller` resources in the `openshift-ingress-operator`
+namespace.
+
+To edit an existing ingress controller:
+
+```shell
+$ oc edit --namespace=openshift-ingress-operator ingresscontroller/<name>
+```
+
+**Important:** Updating an ingress controller may lead to disruption for public
+facing network connections as a new ingress controller revision may be rolled
+out.
+
+Refer to the [ingresscontroller API]() for full details on defaults and
+customizing an ingress controller. The most important initial customizations are
+domain and endpoint publishing strategy, as they *cannot currently be changed
+after the ingress controller is created*.
+
+### Endpoint publishing
+
+The `.spec.endpointPublishingStrategy` field is used to publish the ingress
+controller endpoints to other networks, enable load balancer integrations, etc.
+
+Every strategy is described in detail in the [ingresscontroller API](). A brief
+design diagram for each is shown below.
 
+#### LoadBalancerService
 
-## Reporting issues
+The [LoadBalancerService]() strategy publishes an ingress controller using a
+Kubernetes [LoadBalancer
+Service](https://kubernetes.io/docs/concepts/services-networking/#loadbalancer)
+and on some platforms offers managed wildcard DNS.
 
-Bugs are tracked in [Bugzilla](https://bugzilla.redhat.com/enter_bug.cgi?product=OpenShift%20Container%20Platform&version=4.0.0&component=Routing).
+![Image of LoadBalancerService](docs/images/endpoint-publishing-loadbalancerservice.png)
+
+#### HostNetwork
+
+The [HostNetwork]() strategy uses host networking to publish the ingress
+controller on node ports where the ingress controller is deployed.
```

Comment:
The reader could interpret "on node ports" to mean "using a nodeport service".  How about "directly on the node host where the ingress controller is deployed"?

Reply:
Agree... although we might be able to hold off on this strategy entirely for 4.0, need to discuss tomorrow.